src/clients/rust/tests/tests.rs

```diff
@@ -0,0 +1,302 @@
+use anyhow::Result as AnyResult;
+use futures::executor::block_on;
+use std::process::{Child, Command};
+use std::sync::atomic::{AtomicU16, Ordering};
+use std::{env, fs};
+use tempfile::TempDir;
+
+use tigerbeetle as tb;
+
+const FIRST_PORT_NUMBER: u16 = 20_000;
+static NEXT_PORT_NUMBER: AtomicU16 = AtomicU16::new(FIRST_PORT_NUMBER);
+
+struct TestHarness {
+    tigerbeetle_bin: String,
+    temp_dir: TempDir,
+    port_number: u16,
+    server: Option<Child>,
+}
+
+impl TestHarness {
+    fn new(name: &str) -> AnyResult<TestHarness> {
+        let manifest_dir = env!("CARGO_MANIFEST_DIR");
+
+        let tigerbeetle_bin = if cfg!(unix) {
+            format!("{manifest_dir}/../../../tigerbeetle")
+        } else if cfg!(windows) {
+            format!("{manifest_dir}/../../../tigerbeetle.exe")
+        } else {
+            todo!()
+        };
+
+        let work_dir = format!("{manifest_dir}/work");
+        fs::create_dir_all(&work_dir)?;
+        let temp_dir = TempDir::with_prefix_in(name, &work_dir)?;
```

Comment:
Could use `CARGO_TARGET_TMPDIR` here:

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

Reply:
Done.